tests/unit/src/main/scala/tests/BaseWorksheetLspSuite.scala

```diff
@@ -77,7 +77,7 @@ abstract class BaseWorksheetLspSuite(scalaVersion: String)
            |import java.nio.file.Files
            |val name = "Susan" // "Susan"
            |val greeting = s"Hello $name" // "Hello Susan"
-           |println(greeting + "\nHow are you?") // Hello Susan
+           |println(greeting + "\nHow are you?") // Hello Su...
```

Comment:
So this is debatable whether or not this is good. On one side, I really like it. It shows that there is something more than you're seeing, causing the user to hover on it and indeed see that it goes 2 lines. Even though this is a short line, without it, the user might not do the hover. I'd be curious of your thoughts on it.

Reply:
what is an argument against `// Hello Susan...`? appending 3 chars to the first line without dropping anything?
Can lengthening a line causing any troubles here?

Reply:
So in this exact situation, there is no trouble that can happen by appending it. However, the reason I went for that in general is so that when we hit the cutoff of how many line we display (for very long evaluations), then it won't make them _even_ longer. So the idea would be that even if it's cutoff, the `...` won't go past the limit that we've set.

However, by doing that, I didn't special case it for situations like the test case where we are manually breaking the line. Tbh, I've actually never personally did this in a worksheet, but have came across the long lines version of it many times. So I didn't even think about this situation until my test failed.

Reply:
aaaa ok we manually assume screenWidth  `settings.screenWidth` and we already have truncated line here during processing :D

![2020-04-02-085304_1116x107_scrot](https://user-images.githubusercontent.com/10478402/78219267-7b093d00-74bf-11ea-849f-ea55b4c4961f.png)

About those dots I thought we could in second case append `...` after end if it won't exceed `screenWidth` what do you think?

I think this could be small improvement but if it is not something quick to do just ignore it :+1:

Reply:
Yea that does make sense. Let me see if I can change that. One thing I'm hitting right away is that is that I'm not actually seeing `screenWidth` make any difference. At first I thought that it was due to `coc-metals`, but even in intelliJ whether I have vs code small or full screen, it makes no difference and cuts at the same place. Am I missing something with that setting?

<img width="768" alt="Screenshot 2020-04-02 at 10 21 31" src="https://user-images.githubusercontent.com/13974112/78226259-be69a880-74cb-11ea-87e9-80f352fc33ac.png">

For example in the image, is that suppose to be cutting it off before the end?

Reply:
Ahhh yea, it looks like we don't pass in any screen information.

https://github.com/scalameta/metals/blob/37f4c238c0cc7d83f3362f7269cf63b2c7a4175d/metals/src/main/scala/scala/meta/internal/worksheets/WorksheetProvider.scala#L270-L275

Reply:
We could add a `withScreenWidth`, and `withScreenHeight` here.

Reply:
I don't think vscode uses any truncation mechanisms here, all truncation is done in mdoc according to this:
https://github.com/scalameta/mdoc/blob/master/mdoc/src/main/scala/mdoc/internal/cli/Settings.scala#L136

`screenWidth` I think is a constant and it is not real width of vscode/vim screen.

check this method in mdoc:
`private def renderDecoration(statement: Statement): EvaluatedWorksheetStatement = {`

maybe this truncation should actually happend in mdoc? :question:

Reply:
@olafurpg made a good point about how clients may display truncation differently, so it's probably best for clients to handle it, which makes sense. 

Looking more into this, our test never hit this in the past because before when doing worksheets in vscode, we never added `...` like we did in the editors that didn't support virtual text. So if we simply add that, those tests will fail, since the only check we are doing is to see if the evaluation is complete. I also think this is the only check we can sensibly do as well without getting a whole bunch more information.

That said, the more I think about it, the more I think this is probably valid behavior.

Reply:
Yeah you are right truncation should not be done in `mdoc` 
User will see an output is truncated and that's what is important.

As my after thought we could also do:
adding `margin: Int` to `EvaluatedWorksheetStatement` and in `truncatify`:
```scala
      if (statement.summary().length +3 < statement.margin) statement.summary() + "..."
      else statement.summary().dropRight(3) + "..."
```
But this is a change in mdoc and it makes it more complicated than it should be.